evdns.c

```diff
@@ -4608,21 +4609,20 @@ evdns_nameserver_ip_add_line(struct evdns_base *base, const char *ips) {
 	return 0;
 }
 
-typedef DWORD(WINAPI *GetNetworkParams_fn_t)(FIXED_INFO *, DWORD*);
+typedef DWORD(WINAPI *GetAdaptersAddresses_fn_t)(ULONG, ULONG, PVOID, PIP_ADAPTER_ADDRESSES, PULONG);
 
-/* Use the windows GetNetworkParams interface in iphlpapi.dll to */
+/* Use the windows GetAdaptersAddresses interface in iphlpapi.dll to */
 /* figure out what our nameservers are. */
-static int
-load_nameservers_with_getnetworkparams(struct evdns_base *base)
+int
+load_nameservers_with_getadaptersaddresses(struct evdns_base *base)
 {
-	/* Based on MSDN examples and inspection of  c-ares code. */
-	FIXED_INFO *fixed;
+	PIP_ADAPTER_ADDRESSES addresses = NULL;
 	HMODULE handle = 0;
-	ULONG size = sizeof(FIXED_INFO);
+	ULONG size = sizeof(IP_ADAPTER_ADDRESSES);
 	void *buf = NULL;
-	int status = 0, r, added_any;
-	IP_ADDR_STRING *ns;
-	GetNetworkParams_fn_t fn;
+	int status = 0, r, added_any = 0;
+	GetAdaptersAddresses_fn_t fn;
+	IP_ADAPTER_DNS_SERVER_ADDRESS *dnsserver = NULL;
 
 	ASSERT_LOCKED(base);
```

Comment:
Also this assertion fails now, since the evdns base is not locked in tests

>[err] D:\a\libevent\libevent\evdns.c:4627: Assertion evthread_is_debug_lock_held_((base)->lock) failed in load_nameservers_with_getadaptersaddresses

https://github.com/libevent/libevent/actions/runs/11227695142/job/31214914757?pr=1701

So let's expose the version that holds the lock, i.e. `load_nameservers_with_getadaptersaddresses`, that will internally use `load_nameservers_with_getadaptersaddresses_unlocked` (so as in `evdns_base_config_windows_nameservers`)